src/plugins/graph/utilities/graph-utils.ts

```diff
@@ -140,37 +141,203 @@ export interface IMatchCirclesProps {
+};
+
+export interface ISetPointSelection {
+  dotsRef: IDotsRef
+  dataConfiguration: IDataConfigurationModel
+  pointRadius: number,
+  selectedPointRadius: number,
+  pointColor: string,
+  pointStrokeColor: string,
+  getPointColorAtIndex?: (index: number) => string
+}
+
+
+export function setPointSelection(props: ISetPointSelection) {
+  const { dotsRef, dataConfiguration } = props;
+  const dataset = dataConfiguration.dataset;
+  const allCircles = selectAllCircles(dotsRef.current); //includes both inner and outer circles
+  const outerCircles = selectDots(dotsRef.current, true);
+  if (allCircles){
+    applySelectedClassToCircles(allCircles, dataset);
+  }
+  if (!(dotsRef.current && outerCircles)) return;
+  styleOuterCircles(outerCircles, dataset);
+}
+
+function applySelectedClassToCircles(selection: DotSelection, dataset?: IDataSet){
+  selection
+    .classed('selected', (aCaseData: CaseData) => !!dataset?.isCaseSelected(aCaseData.caseID));
+}
+
+function styleOuterCircles(outerCircles: any, dataset?: IDataSet){
+  if (dataset?.selection.size === 0) {
+    outerCircles
+      .attr('r', (aCaseData: CaseData) => {
+        return 0;
+      });
+  }
+  else {
+    outerCircles
+      .attr('r', (aCaseData: CaseData) => {
+        return (dataset?.isCaseSelected(aCaseData.caseID)) ? 10 : 0;
+      })
+      .style('fill', (aCaseData: CaseData) => {
+        return (dataset?.isCaseSelected(aCaseData.caseID)) && selectedOuterCircleFillColor;
+      })
+      .style('stroke', (aCaseData: CaseData) =>{
+        return selectedOuterCircleStrokeColor;
+      })
+      .style('opacity', 0.5);
+  }
+}
+
+
+export interface ISetPointCoordinates {
+  dataset?: IDataSet
+  dotsRef: IDotsRef
+  selectedOnly?: boolean
+  pointRadius: number
+  selectedPointRadius: number
+  pointColor: string
+  pointStrokeColor: string
+  getPointColorAtIndex?: (index: number) => string
+  getScreenX: ((anID: string) => number | null)
+  getScreenY: ((anID: string, plotNum?:number) => number | null)
+  getLegendColor?: ((anID: string) => string)
+  enableAnimation: React.MutableRefObject<boolean>
+}
+
+export function setPointCoordinates(props: ISetPointCoordinates) {
+
+  const setPoints = (radius: number) => {
+    if (theSelection?.size()) {
+      theSelection
+        .attr('cx', (aCaseData: CaseData) => {
+          return getScreenX(aCaseData.caseID);
+        })
+        .attr('cy', (aCaseData: CaseData) => {
+          return getScreenY(aCaseData.caseID, aCaseData.plotNum);
+        })
+        .attr('r', (aCaseData: CaseData) => {
+          return radius;
+        })
+        .style('fill', (aCaseData: CaseData) => {
+          return lookupLegendColor(aCaseData);
+        })
+        .style('stroke', (aCaseData: CaseData) =>{
+          if (getPointColorAtIndex){
+            return getPointColorAtIndex(aCaseData.plotNum);//border color of inner dot should be same color as legend
```

Comment:
Should this be `lookupLegendColor(aCaseData)`?

Reply:
They both return the same thing, but I'll swap it with your suggestion.